### PR TITLE
N thomas refactor

### PR DESCRIPTION
**Pull request by nathangthomas**
This PR does the following: 
- Adds a partial for vehicle new and edit forms
- Adds skylight to test for application performance in production
- Adds a home link to menu
- Adds button styling to trip index page
- Bolsters testing coverage 
- Adds spacing to user sign up page for better readability

### CODE REVIEW

File: app/views/vehicles/new.html.erb [hunk 1]
```diff
@@ -1,36 +1 @@
...
-  <div class="field">
-    <%= form.label :storage_rack %>
-    <%= form.check_box :storage_rack %>
-  </div>
-
-  <div class="actions">
-    <%= form.submit "Add Vehicle" %>
-  </div>
-<% end %>
+<%= render partial: 'form', locals: {path: '/vehicles', method: :post, button_text: 'Add Vehicle'} %>
```
Comment (hsmitha26) [inline app/views/vehicles/new.html.erb:1]: I will add an edge case test to make sure that the new vehicle is not associated with another user.

File: app/views/vehicles/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<%= form_with model: [:user, @vehicle], local: true do |form| %>
```
Comment (hsmitha26) [inline app/views/vehicles/_form.html.erb:1]: Thanks for refactoring into a partial.

File: .gitignore [hunk 3]
```diff
@@ -28,4 +28,5 @@
 
 # Ignore application configuration
 /config/application.yml
+/config/skylight.yml
 /coverage
```
Comment (leiyakenney) [inline .gitignore:32]: Thanks for installing this!

File: Gemfile [hunk 4]
```diff
@@ -11,6 +11,7 @@ gem 'travis'
 gem 'rake'
 gem 'omniauth-google-oauth2'
 gem 'cancan'
+gem "skylight"
```
Comment (leiyakenney) [inline Gemfile:15]: Thanks for installing this!

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -34,12 +34,16 @@
           <li><strong><i class="far fa-snowflake"></i>
             <%= link_to "About", "#"%></strong></li>
 
-          <li><strong><i class="fas fa-car-side"></i>
-            <%= link_to "My Rides", "#"%></strong></li>
+          <li><strong><i class="fas fa-home"></i>
+            <%= link_to "Home", trip_dashboard_path%></strong></li>
```
Comment (leiyakenney) [inline app/views/layouts/application.html.erb:39]: Good call adding a "Home" here

File: app/views/trips/buttons/index.html.erb [hunk 6]
```diff
@@ -1,11 +1,16 @@
+<div class='ride-buttons'>
 
-<section class='need_ride_button'>
-  <%= link_to "Looking To Put My Junk In Your Trunk", trip_filter_path %><br>
-  <p>Need A Ride</p>
-</section>
+  <section class='need_ride_button'>
+    <strong><%= link_to "Need A Ride", trip_filter_path %></strong><br>
+    <p>(Looking To Put My Junk In Your Trunk)</p>
```
Comment (leiyakenney) [inline app/views/trips/buttons/index.html.erb:5]: Good call swapping the colloquial with the more professional/descriptive

File: app/views/vehicles/_form.html.erb (hunk 2 above)
Comment (leiyakenney) [reply app/views/vehicles/_form.html.erb:1]: Agreed - good call!

File: app/views/trips/buttons/index.html.erb [hunk 7]
```diff
@@ -1,11 +1,16 @@
+<div class='ride-buttons'>
```
Comment (hsmitha26) [inline app/views/trips/buttons/index.html.erb:1]: Great idea!

**Review by leiyakenney — APPROVED** [review]

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -34,12 +34,16 @@
           <li><strong><i class="far fa-snowflake"></i>
```
Comment (hsmitha26) [inline app/views/layouts/application.html.erb]: Home icon and text make more sense, this is great!

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -34,12 +34,16 @@
...
             <%= link_to "About", "#"%></strong></li>
 
-          <li><strong><i class="fas fa-car-side"></i>
-            <%= link_to "My Rides", "#"%></strong></li>
+          <li><strong><i class="fas fa-home"></i>
+            <%= link_to "Home", trip_dashboard_path%></strong></li>
```
Comment (hsmitha26) [inline app/views/layouts/application.html.erb:38]: Home icon and text make more sense, this is great!

File: app/assets/stylesheets/application.css.scss [hunk 10]
```diff
@@ -21,6 +21,14 @@
  */
 
  /* ____________________________________ */
+.field{
+  padding: 10px;
+}
+
+.field label{
+  margin: 0px 0px 1px;
```
Comment (hsmitha26) [inline app/assets/stylesheets/application.css.scss:29]: This will improve the user experience, thank you.

File: Gemfile [hunk 11]
```diff
@@ -11,6 +11,7 @@ gem 'travis'
...
```
Comment (hsmitha26) [inline Gemfile:14]: Looking forward to seeing this in action.
